### PR TITLE
[Enchance] judge `metainfo` is lower

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmyolo/utils/misc.py [hunk 1]
```diff
@@ -114,3 +114,20 @@ def show_data_classes(data_classes):
     # Align display data to the left
     data_classes_info.align['Class name'] = 'l'
     print(data_classes_info)
+
+
+def judge_metainfo_is_lower(cfg):
```
Comment (RangeKing) [inline mmyolo/utils/misc.py:119]:
```suggestion
def is_metainfo_lower(cfg):
```
Would it be better?

File: mmyolo/utils/misc.py [hunk 2]
```diff
@@ -114,3 +114,20 @@ def show_data_classes(data_classes):
...
+    """Determine whether the custom metainfo fields are all lowercase."""
+
+    def judge_keys(dataloader_cfg):
+        while 'dataset' in dataloader_cfg:
```
Comment (triple-Mu) [inline mmyolo/utils/misc.py:123]: dataloader_cfg = dataloader_cfg.get('dataset', {})

File: mmyolo/utils/misc.py (hunk 1 above)
Comment (hhaAndroid) [reply mmyolo/utils/misc.py:119]: OK

File: mmyolo/utils/misc.py [hunk 3]
```diff
@@ -114,3 +114,20 @@ def show_data_classes(data_classes):
...
+def is_metainfo_lower(cfg):
+    """Determine whether the custom metainfo fields are all lowercase."""
+
+    def judge_keys(dataloader_cfg):
+        while 'dataset' in dataloader_cfg:
+            dataloader_cfg = dataloader_cfg['dataset']
+        if 'metainfo' in dataloader_cfg:
+            all_keys = dataloader_cfg['metainfo'].keys()
+            all_is_lower = all([str(k).islower() for k in all_keys])
+            assert all_is_lower, f'The key in dataset metainfo must be all lowercase, but got {all_keys}. ' \
```
Comment (RangeKing) [inline mmyolo/utils/misc.py:128]:
```suggestion
            assert all_is_lower, f'The keys in dataset metainfo must be all lowercase, but got {all_keys}. ' \
```

**Review by RangeKing — APPROVED** [review]

**Pull request merged**